index.js

```diff
@@ -762,6 +775,35 @@ class GithubScm extends Scm {
             sha: pullRequestInfo.head.sha
         }));
     }
+
+    /**
+     * Get an array of scm context (e.g. github.com)
+     * @method getScmContexts
+     * @return {Array}
+     */
+    _getScmContexts() {
+        const contextName = this.config.gheHost
+            ? [`github:${this.config.gheHost}`]
+            : ['github.com'];
+
+        return contextName;
+    }
+
+    /**
+     * Determine if a scm module can handle the received webhook
+     * @method canHandleWebhook
+     * @param {Object}    headers    The request headers associated with the webhook payload
+     * @param {Object}    payload    The webhook payload received from the SCM service
+     * @return {Promise}
+     */
+    _canHandleWebhook(headers, payload) {
+        return this._parseHook(headers, payload)
```

Comment:
To distinguish some scm-githubs, I use not only `_parseHook` but also `gheHost` to check if received headers and payload belong this scm. And this added logic cannot implement in `scm-router` because the format of checkoutUrl is different in each scm.